### PR TITLE
pythonPackages.aplpy: init at 2.0.3

### PR DESCRIPTION
**Pull request by smaret**
###### Motivation for this change

Add a Nix expression for the [Aplpy](http://aplpy.github.io) Python package and its dependencies.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/aplpy/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{ pkgs
```
Comment (dotlambda) [inline pkgs/development/python-modules/aplpy/default.nix]:
```suggestion
{ lib
```

File: pkgs/development/python-modules/aplpy/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+, astropy
+, matplotlib
+, reproject
+}:
+
+buildPythonPackage rec {
+  pname = "aplpy";
+  version = "2.0rc2";
+
+  doCheck = false; # the tests requires several pytest plugins that are not in nixpkgs
```
Comment (dotlambda) [inline pkgs/development/python-modules/aplpy/default.nix]: Can you point me to where you got this info from?

File: pkgs/development/python-modules/astropy-healpix/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+, fetchPypi
+, numpy
+, astropy
+}:
+
+buildPythonPackage rec {
+  pname = "astropy-healpix";
+  version = "0.4";
+
+  doCheck = false; # the tests requires several pytest plugins that are not in nixpkgs
```
Comment (dotlambda) [inline pkgs/development/python-modules/astropy-healpix/default.nix]: Which ones?

File: pkgs/development/python-modules/reproject/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+, astropy
+, astropy-healpix
+, scipy
+}:
+
+buildPythonPackage rec {
+  pname = "reproject";
+  version = "0.4";
+
+  doCheck = false; # the tests requires several pytest plugins that are not in nixpkgs
```
Comment (dotlambda) [inline pkgs/development/python-modules/reproject/default.nix]: Which ones?

**Review by dotlambda — CHANGES_REQUESTED** [review]
Please correct the commit messages and fix the evaluation errors.

File: pkgs/development/python-modules/aplpy/default.nix (hunk 2 above)
Comment (smaret) [reply pkgs/development/python-modules/aplpy/default.nix]: From the build log:
```
RuntimeError: Test dependencies are missing. You should install the 'pytest-astropy' package.
builder for '/nix/store/12ni7z8piw98wqjmjjyvcldf20xwzm2g-python3.7-aplpy-2.0rc2.drv' failed with exit code 1
error: build of '/nix/store/12ni7z8piw98wqjmjjyvcldf20xwzm2g-python3.7-aplpy-2.0rc2.drv' failed
```

File: pkgs/development/python-modules/astropy-healpix/default.nix (hunk 3 above)
Comment (smaret) [reply pkgs/development/python-modules/astropy-healpix/default.nix]: `pytest-astropy` and its dependencies (`pytest-doctestplus`, `pytest-remotedata`, `pytest-openfiles` and `pytest-arraydiff`).

File: pkgs/development/python-modules/reproject/default.nix (hunk 4 above)
Comment (smaret) [reply pkgs/development/python-modules/reproject/default.nix]: `pytest-astropy` and its dependencies (`pytest-doctestplus`, `pytest-remotedata`, `pytest-openfiles` and `pytest-arraydiff`).

File: pkgs/development/python-modules/aplpy/default.nix (hunk 2 above)
Comment (dotlambda) [reply pkgs/development/python-modules/aplpy/default.nix]: Please mention the concrete package that's missing in the comment.

Comment (dotlambda) [reply pkgs/development/python-modules/aplpy/default.nix]: Of course, the optimal thing to do would packaging the dependencies which should not be too difficult.

Comment (dotlambda) [reply pkgs/development/python-modules/aplpy/default.nix]: You can use https://github.com/costrouc/nixpkgs-dev-tools to make it easier.

Comment (smaret) [reply pkgs/development/python-modules/aplpy/default.nix]: I would rather do that in a separate PR.

File: pkgs/development/python-modules/reproject/default.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    inherit pname version;
+    sha256 = "dbbb18a8b211292c7ce61121b8538fc279540337be1c05cabc7570c5aca6d734";
+  };
+
+  propagatedBuildInputs = [ numpy astropy astropy-healpix scipy ];
+
+  meta = with lib; {
+    description = "Reproject astronomical images";
+    homepage = https://reproject.readthedocs.io;
+    license = licenses.bsdOriginal;
```
Comment (dotlambda) [inline pkgs/development/python-modules/reproject/default.nix]: https://github.com/astropy/reproject/blob/master/LICENSE.md is bsd3

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.aplpy python3.pkgs.aplpy

**Comment by dotlambda** [comment]
> Download error on https://pypi.python.org/simple/astropy-helpers/: [Errno -2] Name or service not known -- Some packages may not be found!

File: pkgs/development/python-modules/reproject/default.nix (hunk 5 above)
Comment (smaret) [reply pkgs/development/python-modules/reproject/default.nix]: Weird, it's listed as bsd original on [Pypi](https://pypi.org/project/reproject/).

Comment (dotlambda) [reply pkgs/development/python-modules/reproject/default.nix]: "BSD" can mean anything :)

**Comment by smaret** [comment]
> Download error on https://pypi.python.org/simple/astropy-helpers/: [Errno -2] Name or service not known -- Some packages may not be found!

Where does this come from?

**Comment by dotlambda** [comment]
You will find it in the build log (see the checks):
@GrahamcOfBorg build python2.pkgs.aplpy python3.pkgs.aplpy
It's probably a result of building in the sandbox

**Comment by dotlambda** [comment]
Please reorder the commits and squash the last 3 ones into your earlier ones.

**Comment by smaret** [comment]
@GrahamcOfBorg build python2.pkgs.aplpy python3.pkgs.aplpy

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python3.pkgs.aplpy

**Comment by dotlambda** [comment]
I still get the same error locally.

**Comment by smaret** [comment]
I'm trying to reproduce this error on macOS with sandboxing, but I get a different error:

```
% nix-build -I. -A python3Packages.aplpy
these derivations will be built:
  /nix/store/x69p3bkr0dw7s04zvc1mnyh16qmsrxfb-python3.7-aplpy-2.0.3.drv
building '/nix/store/x69p3bkr0dw7s04zvc1mnyh16qmsrxfb-python3.7-aplpy-2.0.3.drv'...
unpacking sources
unpacking source archive /nix/store/kydlzzhjm9d9a838gqbnc50h7wdzk6fi-APLpy-2.0.3.tar.gz
dyld: Library not loaded: @rpath/CoreFoundation.framework/Versions/A/CoreFoundation
  Referenced from: /nix/store/yaxqbxpg4svc97ic2s01gqsd6d5209ar-gnutar-1.31/bin/tar
  Reason: no suitable image found.  Did find:
        /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation: file system sandbox blocked stat()
/nix/store/zchs18skjv75qg2iy0srwz1dvrrmfkv7-stdenv-darwin/setup: line 842: 83909 Abort trap: 6           tar xf "$fn"
do not know how to unpack source archive /nix/store/kydlzzhjm9d9a838gqbnc50h7wdzk6fi-APLpy-2.0.3.tar.gz
builder for '/nix/store/x69p3bkr0dw7s04zvc1mnyh16qmsrxfb-python3.7-aplpy-2.0.3.drv' failed with exit code 1
error: build of '/nix/store/x69p3bkr0dw7s04zvc1mnyh16qmsrxfb-python3.7-aplpy-2.0.3.drv' failed
```

Is this a bug in the `gnutar` package?

**Comment by dotlambda** [comment]
@smaret I have no experience whatsoever with Darwin. Maybe @matthewbauer can help out.

**Comment by smaret** [comment]
I've just on opened an issue for the gnutar sandbox issue on Darwin (#56591).

**Comment by smaret** [comment]
@GrahamcOfBorg build python3.pkgs.aplpy

**Comment by smaret** [comment]
@GrahamcOfBorg build python3.pkgs.aplpy

**Comment by smaret** [comment]
I finally managed to get the sandbox working on Darwin 🎉

The download error was caused by an automatic update of the `astropy-helpers` module during the setup. I've patched the `setup.cfg` to fix it.

**Comment by smaret** [comment]
@dotlambda Can you please have a look at this when you get a chance? I think that the PR is now ready for merging.

File: pkgs/development/python-modules/astropy-helpers/default.nix [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+  doCheck = false; # tests requires sphinx-astropy
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "37caf1f21bfdf653f7bb9f5b070dc1bb59cd70c0e09f9c5742401f57400a6e52";
+  };
+
+  meta = with lib; {
+    description = "Utilities for building and installing Astropy, Astropy affiliated packages, and their respective documentation.";
```
Comment (dotlambda) [inline pkgs/development/python-modules/astropy-helpers/default.nix]: https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/astropy-helpers/default.nix (hunk 6 above)
Comment (smaret) [reply pkgs/development/python-modules/astropy-helpers/default.nix]: I've removed the period at the end of `meta.description`.

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python3.pkgs.aplpy

**Comment by dotlambda** [comment]
During the build of pyavm I get

> Ran 0 tests in 0.000s

Please add an appropriate `checkPhase` or, if there are no tests, set `doCheck = false` and add a comment with the reason.

**Comment by smaret** [comment]
> During the build of pyavm I get
> 
> > Ran 0 tests in 0.000s
> 
> Please add an appropriate `checkPhase`

Done.

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python3.pkgs.aplpy

**Comment by smaret** [comment]
Thanks for your detailed review of this PR @dotlambda !

**Pull request merged**